lib/mqtt.c

```diff
@@ -287,7 +296,12 @@ static CURLcode mqtt_connect(struct Curl_easy *data)
   result = Curl_rand_hex(data, (unsigned char *)&client_id[clen],
                          MQTT_CLIENTID_LEN - clen + 1);
   /*adding client id*/
-  add_client_id(client_id, packet, pos + 1);
+  rc = add_client_id(client_id, strlen(client_id), packet, pos + 1);
+  if(rc) {
+    infof(data, "Client ID length mismatched: [%lu]\n", strlen(client_id));
```

Comment:
Use `failf()` for when you return error. And then you need to remove the newline. Goes for the other cases below too!

Reply:
Thanks @bagder for the patience 👍